component/otelcol/exporter/exporter.go

```diff
@@ -69,7 +71,7 @@ var (
 //
 // The registered component must be registered to export the
 // otelcol.ConsumerExports type, otherwise New will panic.
-func New(opts component.Options, f otelexporter.Factory, args Arguments) (*Exporter, error) {
+func New(opts component.Options, f otelexporter.Factory, args Arguments, disableMetricsExporter bool) (*Exporter, error) {
```

Comment:
I think adding a `disableMetricsExporter` parameter feels a little too targeted for this specific bug. Since the exporter package is not where the bug originates, I think it can make adding this specific parameter a big confusing and vague. 

I'd like to see this fix transform into a more general improvement to the exporter package's API, such as some kind of bitfield for which telemetry types to create exporters for, such as calling:

* `exporter.New(opts, fact, args.(Arguments), exporter.TypesAll)`, or
* `exporter.New(opts, fact, args.(Arguments), exporter.TypeLogs|exporter.TypeTraces)`. 

This will be more work, but I think it leaves the API design feeling more intentional rather than having the ability to disable specifically metrics.

Reply:
TBH I'm hoping we just get rid of this workaround soon altogether. We shouldn't be initialising a metrics exporter if other components are not configured to send metrics to the LB exporter. I think that's the real problem.

Reply:
I added a bitfield just now, and also tested it locally by running the agent with an LB exporter and an otlp one. PTAL.